core/index.js

```diff
@@ -69,31 +69,83 @@ DattCore.prototype.initialize = function () {
  */
 DattCore.prototype.asyncInitialize = function () {
   return asink(function *() {
+    yield this.asyncInitializeDb()
+    yield this.asyncInitializeCoreContent()
+
+    yield this.asyncInitializeCoreUser()
+    yield this.asyncInitializeCoreBitcoin()
+    yield this.asyncInitializeCorePeers()
+
+    this.isinitialized = true
+    this.emit('initialized', this.isinitialized)
+    return Promise.resolve()
```

Comment:
Inside asink you can just return, and it will return a promise automatically. You don't need to return Promise.resolve(). This is the same as with the async/await pattern.